src/index.js

```diff
@@ -1,174 +1,231 @@
 'use strict'
 
-import createWildcard from './wild'
+import createWildcard from './wildcard'
 
 const isArray = Array.isArray
 const keys = Object.keys
 
-export default options => {
+export default (options = {}) => {
   const wild = createWildcard({ enabled: options.wildcard })
 
-  let rooms = {}
-  let sids = {}
+  let rooms = new Map();
+  let sids = new Map();
 
   /**
-   * Get rooms socket is subscribed to.
-   *
-   * @param {String} id
-   * @param {Function} [fn]
+   * Get rooms socket is subscribed to. Or all rooms in the server
+   * Used for get rooms
+   * @param {String} [id] - spark's ID
+   * @returns {Promise} containing rooms the s
    * @api public
    */
-
-  function get(id, fn = noop) {
-    let rms = keys(id ? sids[id] || {} : rooms)
-    setImmediate(fn, null, rms)
+  function get(id) {
+    if (id) {
+      return Promise.resolve(Array.from(sids.get(id)));
+    }
+    let roomsArray = [];
+    rooms.forEach((value, key) => {
+      roomsArray.push(key);
+    });
+    return Promise.resolve(roomsArray);
   }
 
   /**
    * Add a socket to a room.
-   *
-   * @param {String} id
-   * @param {String} room
-   * @param {Function} [fn]
+   * Used for joining rooms
```

Comment:
I still find these redundant. "Add a socket to a room" - "Used for joining rooms" yeah if it adds a socket to room it is obviously used for joining rooms :)